Assets/VRTK/Scripts/VRTK_MoveInPlace.cs

```diff
@@ -0,0 +1,387 @@
+        public float GetSpeed()
+        {
+            return curSpeed;
+        }
+
+        private void Awake()
+        {
+            controllerLeftHand = VRTK_DeviceFinder.GetControllerLeftHand();
+            controllerRightHand = VRTK_DeviceFinder.GetControllerRightHand();
+            headset = VRTK_DeviceFinder.HeadsetTransform();
+
+            SetControlOptions(controlOptions);
+        }
+
+        private void OnEnable()
+        {
+            touchpadPressed += DoTouchpadDown;
+            touchpadUp += DoTouchpadUp;
+        }
+
+        private void OnDisable()
+        {
+            touchpadPressed -= DoTouchpadDown;
+            touchpadUp -= DoTouchpadUp;
+        }
+
+        private void Start()
+        {
+            rigidBody = GetComponent<Rigidbody>();
+            SetControllerListeners(controllerLeftHand);
+            SetControllerListeners(controllerRightHand);
+
+            // Initialize the lists.
+            foreach (Transform trackedObj in trackedObjects)
+            {
+                movementList.Add(trackedObj, new List<float>());
+                previousYPositions.Add(trackedObj, trackedObj.transform.localPosition.y);
+                cameraLastYRotation = headset.transform.rotation.y;
+            }
+            playAreaPreviousY = transform.position.y;
+        }
+        
+        private void DoTouchpadDown(object sender, ControllerInteractionEventArgs e)
+        {
+            var controllerEvents = (VRTK_ControllerEvents)sender;
+            active = true;
+        }
+
+        private void DoTouchpadUp(object sender, ControllerInteractionEventArgs e)
+        {
+            var controllerEvents = (VRTK_ControllerEvents)sender;
+
+            // If the button is released, clear all the lists.
+            foreach (Transform obj in trackedObjects)
+            {
+                movementList[obj].Clear();
+            }
+            initalGaze = Vector3.zero;
+            direction = Vector3.zero;
+            curSpeed = 0;
+
+            active = false;
+        }
+        
+        private void FixedUpdate()
+        { 
+            // If Move In Place is currently engaged.
+            if (active)
+            {
+                // Initialze the list average.
+                float listAverage = 0;
+
+                // Get the y-position delta of the play area.
+                float playAreaDeltaY = Mathf.Abs(playAreaPreviousY - transform.position.y);
+
+                foreach (Transform trackedObj in trackedObjects)
+                {
+                    // Get the amount of Y movement that's occured since the last update.
+                    float deltaYPostion = Mathf.Abs(previousYPositions[trackedObj] - trackedObj.transform.localPosition.y);
+
+                    // Convenience code.
+                    List<float> trackedObjList = movementList[trackedObj];
+
+                    if ((playerBody != null && (!playerBody.IsFalling())) || playerBody == null)
```

Comment:
playerBody is never set to anything so it will always be null meaning the first chunk of this if statement will never be true.

Should playerBody be initialised somewhere?